Makefile

```diff
@@ -260,6 +268,48 @@ include lib/stack_protector/stack_protector.mk
 
 include ${PLAT_MAKEFILE_FULL}
 
+ifeq (${ARM_ARCH_MAJOR},7)
+# For ARMv7, set march32 from platform directive ARMV7_CORTEX_Ax=yes
+# and ARM_WITH_NEON=yes/no.
+#
+# GCC and Clang require -march=armv7-a for C-A9 and -march=armv7ve for C-A15.
+# armClang requires -march=armv7-a for all ARMv7 Cortex-A. To comply with
+# all, just drop -march and supply only -mcpu.
+#
+# Platform can override march32-directive.
+march32-set-${ARM_CORTEX_A5}	:= -mcpu=cortex-a5
+march32-set-${ARM_CORTEX_A7}	:= -mcpu=cortex-a7
+march32-set-${ARM_CORTEX_A9}	:= -mcpu=cortex-a9
+march32-set-${ARM_CORTEX_A15}	:= -mcpu=cortex-a15
+march32-neon-$(ARM_WITH_NEON)	:= -mfpu=neon
```

Comment:
What about Cortex-A12/A17 [both called A17 nowadays]? I don't think there should be much difference to the others, but maybe we still want to have separate options?

Reply:
Yes, I guess A12/A17 could be supported too.
I will add a change for that.